File: backend/models/Document.js
const mongoose = require("mongoose");

const documentSchema = new mongoose.Schema({

  title: String,

  department: String,

  subject: String,

  category: String,

  fileName: String,

  fileUrl: String,

  uploadedBy: String,

  uploaderId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "User",
    default: null,
  },

  uploaderRole: {
    type: String,
    default: "",
  },

  /** When category is Assignment: teacher handout vs student submission */
  assignmentKind: {
    type: String,
    enum: ["handout", "submission"],
    default: undefined,
  },

  createdAt: {
    type: Date,
    default: Date.now,
  },

});

module.exports =
  mongoose.model(
    "Document",
    documentSchema
  );

File: DotNetBackend/Models/Document.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System.Text.Json.Serialization;

namespace DotNetBackend.Models
{
    [BsonIgnoreExtraElements]
    public class Document
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        [JsonPropertyName("_id")]
        public string? Id { get; set; }

        [BsonElement("title")]
        public string? Title { get; set; }

        [BsonElement("department")]
        public string? Department { get; set; }

        [BsonElement("subject")]
        public string? Subject { get; set; }

        [BsonElement("category")]
        public string? Category { get; set; }

        [BsonElement("fileName")]
        public string? FileName { get; set; }

        [BsonElement("fileUrl")]
        public string? FileUrl { get; set; }

        [BsonElement("uploadedBy")]
        public string? UploadedBy { get; set; }

        [BsonElement("uploaderId")]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? UploaderId { get; set; }

        [BsonElement("uploaderRole")]
        public string? UploaderRole { get; set; }

        [BsonElement("assignmentKind")]
        [BsonIgnoreIfNull]
        public string? AssignmentKind { get; set; }

        [BsonElement("createdAt")]
        public DateTime? CreatedAt { get; set; } = DateTime.UtcNow;
    }
}
